sympy/physics/mechanics/jointsmethod.py

```diff
@@ -0,0 +1,222 @@
+
+    @property
+    def mass_matrix(self):
+        return self.method.mass_matrix
+
+    @property
+    def forcing(self):
+        return self.method.forcing
+
+    @property
+    def method(self):
+        return self._method
+
+    def _generate_bodylist(self):
+        bodies = []
+        for joint in self._joints:
+            if joint.child not in bodies:
+                bodies.append(joint.child)
+            if joint.parent not in bodies:
+                bodies.append(joint.parent)
+        return bodies
+
+    def _generate_loadlist(self):
+        load_list = []
+        for body in self.bodies:
+            load_list.extend(body.loads)
+        return load_list
+
+    def _generate_q(self):
+        q_ind = []
+        for joint in self._joints:
+            q_ind.extend(joint.coordinates)
+        return q_ind
+
+    def _generate_u(self):
+        u_ind = []
+        for joint in self._joints:
+            u_ind.extend(joint.speeds)
+        return u_ind
+
+    def _generate_kdes(self):
+        kd_ind = []
+        for joint in self._joints:
+            kd_ind.extend(joint.kdes)
+        return kd_ind
+
+    def form_eoms(self, method=KanesMethod):
+        """Method to form system's equation of motions.
+
+        Parameters
+        ==========
+
+        method : Class
+            Class name of method.
+
+        Examples
+        ========
+
+        This is a simple example for a one degree of freedom translational
+        spring-mass-damper.
+
+        >>> from sympy.physics.mechanics import LagrangesMethod, dynamicsymbols, Body
+        >>> from sympy.physics.mechanics import PrismaticJoint, JointsMethod
+        >>> from sympy import symbols
+        >>> q = dynamicsymbols('q')
+        >>> qd = dynamicsymbols('q', 1)
+        >>> m, k, b = symbols('m k b')
+        >>> B = Body('B')
+        >>> P = Body('P', mass=m)
+        >>> P.potential_energy = k * q**2 / 2.0
+        >>> J = PrismaticJoint('J', B, P, coordinates=q, speeds=qd)
+        >>> B.apply_force(b * qd * B.x, reaction_body=P)
+        >>> method = JointsMethod(B, J)
+        >>> method.form_eoms(LagrangesMethod)
+        Matrix([[b*Derivative(q(t), t) + 1.0*k*q(t) + m*Derivative(q(t), (t, 2))]])
+
+        We can also solve for the states using the 'rhs' method.
+
+        >>> method.rhs()
+        Matrix([
+        [                    Derivative(q(t), t)],
+        [(-b*Derivative(q(t), t) - 1.0*k*q(t))/m]])
+
+        """
+
+        if issubclass(method, KanesMethod): #KanesMethod or similar
+            self._method = method(self.frame, q_ind=self.q, u_ind=self.u, kd_eqs=self.kdes,
+                                    forcelist=self.forcelist, bodies=self.bodies)
+        if issubclass(method, LagrangesMethod): #LagrangesMethod or similar
+            L = Lagrangian(self.frame, *self.bodies)
+            self._method = method(L, self.q, self.forcelist, self.bodies, self.frame)
+        soln = self.method._form_eoms()
+        return soln
```

Comment:
We should return the same result, if anything. Always return `M*u' + F` or `M*q'' + F`.

Reply:
Now in Kanes it returns fr-fr*